### PR TITLE
Fix #496: Remove build time dependency of power-auth-push-server on running environment

### LINKED ISSUE #496: Remove build time dependency of power-auth-push-server on running environment
remove dependency on powerauth-server in powerauth-push-server junit tests - probably by mocking the dependency

### PR DESCRIPTION
**Pull request by romanstrobl**

### CODE REVIEW

File: powerauth-push-server/src/test/java/io/getlime/push/controller/MockPowerAuthController.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+    @PostMapping("application/create")
+    public ObjectResponse<CreateApplicationResponse> createApplication() {
+        final CreateApplicationResponse response = new CreateApplicationResponse();
+        return new ObjectResponse<>(response);
+    }
+
+    @PostMapping("application/detail")
+    public ObjectResponse<GetApplicationDetailResponse> getApplicationDetail() {
```
Comment (banterCZ) [inline powerauth-push-server/src/test/java/io/getlime/push/controller/MockPowerAuthController.java:63]: I know, it is only a test, but better not to swallow the exception. But what about
```suggestion
    public ObjectResponse<GetApplicationDetailResponse> getApplicationDetail() throw CryptoProviderException {
```

File: powerauth-push-server/src/test/java/io/getlime/push/controller/MockPowerAuthController.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        version.setSupported(true);
+        version.setApplicationKey("key");
+        version.setApplicationSecret("secret");
+        response.getVersions().add(version);
+        try {
+            final KeyPair keyPair = new KeyGenerator().generateKeyPair();
+            final KeyConvertor keyConvertor = new KeyConvertor();
+            response.setMasterPublicKey(BaseEncoding.base64().encode(keyConvertor.convertPublicKeyToBytes(keyPair.getPublic())));
+        } catch (CryptoProviderException e) {
+            // Ignore error in mock endpoint
```
Comment (banterCZ) [inline powerauth-push-server/src/test/java/io/getlime/push/controller/MockPowerAuthController.java:77]: Or at least
```suggestion
            logger.error("Crypto provider error", e);
```

**Review by banterCZ — APPROVED** [review]

File: powerauth-push-server/src/test/java/io/getlime/push/controller/MockPowerAuthController.java (hunk 1 above)
Comment (romanstrobl) [reply powerauth-push-server/src/test/java/io/getlime/push/controller/MockPowerAuthController.java:63]: We don't want to declare an exception that can never happen.

File: powerauth-push-server/src/test/java/io/getlime/push/controller/MockPowerAuthController.java (hunk 2 above)
Comment (romanstrobl) [reply powerauth-push-server/src/test/java/io/getlime/push/controller/MockPowerAuthController.java:77]: Yes, logging is fine, but again, the exception cannot happen.

**Pull request merged**